eks/ng/cluster-autoscaler/cluster-autoscaler.go

```diff
@@ -0,0 +1,396 @@
+subjects:
+  - kind: ServiceAccount
+    name: cluster-autoscaler
+    namespace: kube-system
+---
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cluster-autoscaler
+  namespace: kube-system
+  labels:
+    app: cluster-autoscaler
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: cluster-autoscaler
+  template:
+    metadata:
+      labels:
+        app: cluster-autoscaler
+      annotations:
+        prometheus.io/scrape: 'true'
+        prometheus.io/port: '8085'
+        cluster-autoscaler.kubernetes.io/safe-to-evict: 'false'
+    spec:
+      serviceAccountName: cluster-autoscaler
+      containers:
+{{ if ne .ImageURI "" }}{{.ImageURI}}{{ end }}
+          name: cluster-autoscaler
+          resources:
+            limits:
+              cpu: 100m
+              memory: 300Mi
+            requests:
+              cpu: 100m
+              memory: 300Mi
+          command:
+            - ./cluster-autoscaler
+            - --v=4
+            - --stderrthreshold=info
+            - --cloud-provider=aws
+            - --skip-nodes-with-local-storage=false
+            - --expander=least-waste
+{{ if ne .NodeGroupAutoDiscovery "" }}{{.NodeGroupAutoDiscovery}}{{ end }}
+            - --balance-similar-node-groups
+            - --skip-nodes-with-system-pods=false
+          volumeMounts:
+            - name: ssl-certs
+              mountPath: /etc/ssl/certs/ca-certificates.crt
+              readOnly: true
+          imagePullPolicy: "Always"
+      volumes:
+        - name: ssl-certs
+          hostPath:
+            path: "/etc/ssl/certs/ca-bundle.crt"
+`
+
+var images = map[string]string{
+	"1.17": `        - image: us.gcr.io/k8s-artifacts-prod/autoscaling/cluster-autoscaler:v1.17.2`,
+	"1.16": `        - image: us.gcr.io/k8s-artifacts-prod/autoscaling/cluster-autoscaler:v1.16.5`,
+}
+
+const (
+	nodeGroupAuotDiscoveryData      = `            - --node-group-auto-discovery=asg:tag=k8s.io/cluster-autoscaler/enabled,k8s.io/cluster-autoscaler/`
+	clusterAutoScalerDeploymentName = "cluster-autoscaler"
+)
+
+type caSpecData struct {
+	ImageURI               string
+	NodeGroupAutoDiscovery string
+}
+
+// Config defines version upgrade configuration.
+type Config struct {
+	Logger    *zap.Logger
+	Stopc     chan struct{}
+	EKSConfig *eksconfig.Config
+	K8SClient k8s_client.EKS
+	EC2API    ec2iface.EC2API
+	ASGAPI    autoscalingiface.AutoScalingAPI
+	EKSAPI    eksiface.EKSAPI
+}
+
+// ClusterAutoScaler defines cluster autoscaler operation.
+type ClusterAutoscaler interface {
+	// Wait waits until all NG and Kubernetes nodes are ready.
+	CreateCA() error
+}
+
+// New creates a new cluster autoscaler.
+func New(cfg Config) ClusterAutoscaler {
+	cfg.Logger.Info("creating ClusterAutoscaler", zap.String("tester", reflect.TypeOf(tester{}).PkgPath()))
+	return &tester{cfg: cfg}
+}
+
+type tester struct {
+	cfg Config
+}
+
+func (ts *tester) CreateCA() error {
+	return ts.createCA()
+}
+
+func (ts *tester) createCA() error {
```

Comment:
Can we iterate all `ts.cfg.EKSConfig.AddOnNodeGroups.ASGs` and check if we need install CA?

```
needInstall := false
for ... range ts.cfg.EKSConfig.AddOnNodeGroups.ASGs {
   if cur.Autoscaler.Enable { needInstall = true; break; }
   ...

if !needInstall { lg.Info("no NG enables CA; skipping"); return nil; }
```